yellow-paper/docs/public-vm/avm.md

```diff
@@ -0,0 +1,78 @@
+
+The **["AVM Instruction Set"](./InstructionSet)** document supplements this one with the list of all supported instructions and their associated state transition functions.
+
+> Note: The Aztec Virtual Machine, while designed with a SNARK implementation in mind, is not strictly tied to any particular implementation and therefore is defined without SNARK or circuit-centric verbiage. That being said, considerations for a SNARK implementation are raised or linked when particularly relevant or helpful.
+
+## Public contract bytecode
+A contract's public bytecode is series of execution instructions for the AVM. When a message call is made to a contract, the AVM retrieves the corresponding bytecode from the world state (`worldState.contracts[address].bytecode`) and triggers execution of the first instruction (`bytecode[0]`). The world state is described in more detail later.
+
+> Note: While a Noir contract may have multiple public functions, they are inlined so that the **entirety of a contract's public code exists in a single bytecode**. Internal calls to Noir functions within the same contract are compiled to simple program-counter changes, as are internal returns. In a manner similar to the Ethereum Virtual Machine, the AVM is not itself aware of function selectors and internal function calls. The Noir compiler may implement these constructs by treating the first word in a message call's calldata as a function selector, and beginning a contract's bytecode with a series of conditional jumps.
+
+> Note: See the [Bytecode Validation Circuit](./bytecode-validation-circuit.md) to see how a contract's bytecode can be validated and committed to which is particularly relevant for a SNARK implementation of the AVM.
+
+## Execution Context
+:::note REMINDER
+Many terms and definitions here are borrowed from the [Ethereum Yellow Paper](https://ethereum.github.io/yellowpaper/paper.pdf).
+:::
+
+An "Execution Context" includes the information necessary to trigger AVM execution along with the state maintained by the AVM throughout execution:
+```
+AVMContext {
+    environment: ExecutionEnvironment,
+    machineState: MachineState,
+    worldState: WorldState,
+    accruedSubstate: AccruedSubstate,
+    results: MessageCallResults,
+}
+```
+
+The "Execution Environment" is provided by the execution agent and remains constant throughout a message call's execution. A new environment is initialized for any nested message calls.
+```
+ExecutionEnvironment {
+    address,
+    origin,
+    l1GasPrice,
+    l2GasPrice,
+    calldata: [],
+    sender,
+    portal,
+    bytecode: [],
+    blockHeader,
+    messageCallDepth,
+    isStaticCall,
+}
+```
+> TODO: what about global variables?
+
+"Machine State" belongs to single message call and is transformed on an instruction-per-instruction basis. A message call's machine state is initialized, alongside its execution environment, prior to the start of its execution:
```

Comment:
Could we summarize in one sentence that ExecutionEnvironment and MachineState have the same lifecycle and that they both represent information pertaining to a single message call. The former being the constant part while the former is the variable part?

We might also emphasize whether "a message call execution" here means a single function call or a single function call with all the nested internal calls. Maybe we could clarify this above in the document. I understood that all internal calls are kind of transparent so that the answer is probably the second one.